test_conformance/api/negative_queue.cpp

```diff
@@ -57,55 +168,122 @@ int test_negative_create_command_queue_with_properties(cl_device_id deviceID,
+        "when: \"values specified in properties are not valid\"",
+        TEST_FAIL);
 
-    test_failure_error_ret(test_error, CL_INVALID_QUEUE_PROPERTIES,
-                           "clCreateCommandQueueWithProperties should "
-                           "return CL_INVALID_QUEUE_PROPERTIES if "
-                           "values specified in properties are valid but "
-                           "are not supported by the "
-                           "device.",
-                           TEST_FAIL);
+    cl_command_queue_properties device_queue_properties = 0;
+    err = clGetDeviceInfo(deviceID, CL_DEVICE_QUEUE_PROPERTIES,
+                          sizeof(device_queue_properties),
+                          &device_queue_properties, nullptr);
+    test_error(err, "clGetDeviceInfo");
+    std::initializer_list<cl_command_queue_properties> valid_properties{
+        CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE, CL_QUEUE_PROFILING_ENABLE
+    };
+    properties[0] = CL_QUEUE_PROPERTIES;
+    bool missing_property = false;
+    // Iterate through all possible properties to find one which isn't supported
+    for (auto property : valid_properties)
+    {
+        if ((device_queue_properties & property) == 0)
+        {
+            missing_property = true;
+            properties[1] = property;
+            break;
+        }
+    }
+    if (missing_property)
+    {
+        clCreateCommandQueueWithProperties(context, deviceID, properties.data(),
+                                           &err);
+        test_failure_error_ret(
+            err, CL_INVALID_QUEUE_PROPERTIES,
+            "clCreateCommandQueueWithProperties should return "
+            "CL_INVALID_QUEUE_PROPERTIES when: \"values specified in "
+            "properties are valid but are not supported by the device\"",
+            TEST_FAIL);
+    }
+    else if (get_device_cl_version(deviceID) >= Version(2, 0))
+    {
+        cl_uint max_size = -1;
+        err = clGetDeviceInfo(deviceID, CL_DEVICE_QUEUE_ON_DEVICE_MAX_SIZE,
+                              sizeof(max_size), &max_size, nullptr);
+        test_error(err, "clGetDeviceInfo");
+        if (max_size > 0)
+        {
+            properties[0] = CL_QUEUE_PROPERTIES;
+            properties[1] =
+                CL_QUEUE_ON_DEVICE | CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE;
+            properties[2] = CL_QUEUE_SIZE;
+            properties[3] = max_size + 1;
+            clCreateCommandQueueWithProperties(context, deviceID,
+                                               properties.data(), &err);
+            test_failure_error_ret(
+                err, CL_INVALID_VALUE,
+                "clCreateCommandQueueWithProperties should return "
+                "CL_INVALID_VALUE when: \"values specified in "
+                "properties are not valid\" using a queue size "
+                "greater than CL_DEVICE_QUEUE_ON_DEVICE_MAX_SIZE",
+                TEST_FAIL);
```

Comment:
Spec:
[CL_INVALID_VALUE](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_INVALID_VALUE) if values specified in properties are not valid.

[CL_INVALID_QUEUE_PROPERTIES](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_INVALID_QUEUE_PROPERTIES) if values specified in properties are valid but are not supported by the device.

Looks like this could equally be interpreted as a case for "CL_INVALID_QUEUE_PROPERTIES" since the size itself is "not supported by the device" having exceeded CL_DEVICE_QUEUE_ON_DEVICE_MAX_SIZE. 

Would like to discuss this in the next call.

Reply:
Agreed on 8/27 to accept either error codes (CL_INVALID_VALUE or CL_INVALID_QUEUE_PROPERTIES) for this case. Arm will make this fix in a subsequent patch.

Reply:
@ahesham-arm Looks like the PR will need updating to address the request above.

Reply:
Done.